lib/board.rb

```diff
@@ -50,77 +50,55 @@ def all_equal?(coordinate_array)
     coordinate_array.uniq.size <= 1
   end
 
-    def consecutive_spaces
-      coord_array_setup
-      if all_equal?(@col_array)
-        true if array_increments?(@row_array)
-      elsif array_increments?(@col_array)
-        true if all_equal?(@row_array)
-      else
-        false
-      end
+  def consecutive_spaces
+    coord_array_setup
+    if all_equal?(@col_array)
+      true if array_increments?(@row_array)
+    elsif array_increments?(@col_array)
+      true if all_equal?(@row_array)
+    else
+      false
     end
+  end
 
-
-    def valid_placement?(ship, coordinate_array)
-      @coordinate_array = coordinate_array
-      if different_lengths(ship, coordinate_array)
-        false
-      elsif invalid_coordinate(coordinate_array)
-        false
-      elsif coordinate_already_has_ship(coordinate_array)
-        false
-      else
-        consecutive_spaces
-      end
-
+  def valid_placement?(ship, coordinate_array)
+    @coordinate_array = coordinate_array
+    if different_lengths(ship, coordinate_array)
+      false
+    elsif invalid_coordinate_array(coordinate_array)
+      false
+    elsif coordinate_already_has_ship(coordinate_array)
+      false
+    else
+      consecutive_spaces
     end
-
   end
 
-#   def render_helper
-#     if @cell.ship == nil
-#       p "M"
-#     elsif @cell.ship
-#   end
-#
-#   def render(show_ships = false )
-#     if @cell.fired_upon? == false
-#       if show_ships == true
-#         p "S"
-#       else
-#         p "."
-#       end
-#     else
-#     end
-#   end
-# end
-
-
-    def different_lengths(ship, coordinate_array)
-      ship.length != coordinate_array.length
-    end
+  def different_lengths(ship, coordinate_array)
+    ship.length != coordinate_array.length
+  end
 
-    ######## might consider adding invalid_coordinate_array to distinguish from valid_coordinate
-    def invalid_coordinate(coordinate_array)
-      coordinate_array.any? {|coordinate| !valid_coordinate?(coordinate)}
-    end
+  def invalid_coordinate_array(coordinate_array)
+    coordinate_array.any? {|coordinate| !valid_coordinate?(coordinate)}
+  end
```

Comment:
totally agree with your comment to distinguish from valid_coordinate. Added invalid_coordinate_array